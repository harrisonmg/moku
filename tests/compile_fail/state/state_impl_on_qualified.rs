use moku::*;

#[state_machine]
mod blinky {
    use moku::*;

    #[machine_module]
    mod state_machine {}

    struct Top;
    impl TopState<BlinkyState> for Top {}

    struct Bottom;

    #[superstate(Top)]
    impl State<BlinkyState> for super::blinky::Bottom {}
}

fn main() {}
