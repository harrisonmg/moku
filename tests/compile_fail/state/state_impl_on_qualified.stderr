error: `moku::State` must be implemented on a plain struct
       You may also use a type alias: `type MyState = Option<bool>;`
  --> tests/compile_fail/state/state_impl_on_qualified.rs:16:33
   |
16 |     impl State<BlinkyState> for super::blinky::Bottom {}
   |                                 ^^^^^
